File: Modules/Kit/MES/SIE.Web.Kit.MES/CallMaterials/Commands/ExcuteSolutionCommand.cs
using SIE.Kit.MES.CallMaterials;
using SIE.Web.Command;

namespace SIE.Web.Kit.MES.CallMaterials.Commands
{
    /// <summary>
    /// 执行方案命令
    /// </summary>
    [JsCommand("SIE.Web.Kit.MES.CallMaterials.Commands.ExcuteSolutionCommand")]
    public class ExcuteSolutionCommand : SaveCommand
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">SIE.MetaModel.View.Block.EntityType</param>
        /// <returns>object</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            RT.Service.Resolve<CallMaterialController>().ExcuteSortSolution();
            return "执行成功";
        }
    }
}

File: Modules/Kit/MES/SIE.Web.Kit.MES/CallMaterials/Commands/ExcuteSolutionCommand.js
SIE.defineCommand('SIE.Web.Kit.MES.CallMaterials.Commands.ExcuteSolutionCommand', {
    meta: { text: "执行方案", group: "edit", iconCls: "icon-Implementation icon-blue" },
    execute: function (view, source) {
        var me = this;
        var resourceName = view._relations[0]._target.getData().data.ResourceName;
        var flag = 0;
        var indata = {};
        indata.data = Ext.encode({ Flag: flag, ResourceName: resourceName });
        view.execute({
            data: indata,
            success: function (res) {
                var errMsg = res.Result;
                if (errMsg == '执行成功')
                    view.reloadData();
                    SIE.Msg.showMessage(errMsg);
            }
        });
    }
});